src/script/queue.rs

```diff
@@ -0,0 +1,81 @@
+#[derive(Debug)]
```

Comment:
Any reason for not using [VecDeque](https://doc.rust-lang.org/1.15.0/std/collections/struct.VecDeque.html)? It doesn't have `peek` but it can be easily simulated with `get`